### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by Alejuda**
Hi there! in this branch, we worked pair to pair with my coding partner @MMGGYY66 and did the following:

- [Create Run class and define inicialize method](https://github.com/Alejuda/oop-school-library/commit/ebc5bdab15ff6ee48f72573ead17495cf0dae3fd)
- [Create entry_point methos inside the Run class](https://github.com/Alejuda/oop-school-library/commit/e8b8a28aaf24930bd25d0dbf171f8b993c0bd855)
- [Create options methos inside the Run class](https://github.com/Alejuda/oop-school-library/commit/ffb63353b189e92f72bb37e66e01ba03f1057916)
- Make the main class to only start the code and ask the user for the initial choice of options (or to quit).
- [Run rubocop and fix linters errors](https://github.com/Alejuda/oop-school-library/commit/6521e205602a56a48a009bb74948287a577c1844)
- [Add Mohammed as a collaborator in readme file](https://github.com/Alejuda/oop-school-library/pull/5/commits/1a9dda4174ab09ad64ec6ecb0f095ba66826d0c2)
- [Modifyed MIT license](https://github.com/Alejuda/oop-school-library/pull/5/commits/4862390eb48474e17dd2c0d6c0224f0608417d9b)

### CODE REVIEW

File: run.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    puts '7 - Exit'
+  end
+
+  def entry_point
+    puts 'Welcome to School Library App!'
+
+    loop do
+      show_options
+      app_choice = gets.chomp.to_i
+      options(app_choice)
```
Comment (Ibizugbe) [inline run.rb]: Nice job assigning the responsibility of input validation and request to this class😉. But let's also ensure every responsibility of requesting and validating inputs is done here to obey the single responsibility rule which states that every class should have a single responsibility. This makes our codebase maintainable and easier to understand, hence making it a display of professionalism👍🏼. currently, the functions of validating and requesting inputs are still present in the `app` class😉. let's do well to bring all of them here and call them when necessary in the respective classes. You can take a look at [this lesson ](https://www.digitalocean.com/community/conceptual-articles/s-o-l-i-d-the-first-five-principles-of-object-oriented-design)for reference😉.

Screenshot 2023-07-31 at 17 08 07

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @Alejuda  @MMGGYY66 👋🏼👋🏼👋🏼, 

require_changes


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Nice job assigning the responsibility of requesting and validating input to one class👍🏼.
- [x] Gitflow is used👍🏼.


## Required Changes ♻️

- [ ] Your application is also currently breaking, do well to fix that as the app ought to run well after the code has been refactored👍🏼.
Screenshot 2023-07-31 at 17 30 49


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Alejuda** [comment]
We have corrected all the required changes. Thanks for your feedback!

File: app.rb [hunk 2]
```diff
@@ -12,54 +12,76 @@ def initialize
...
     end
   end
 
+  def create_student
+    print 'Age: '
+    person_age = gets.chomp.to_i
+    print 'Name: '
+    person_name = gets.chomp.to_s
+    print 'Has parent permission? [Y/N]: '
+    person_permission = gets.chomp.to_s.capitalize
```
Comment (tugsem) [inline app.rb]: - [ ] As required by the reviewer, all functions ask the user input should be in the class for getting user data. In your case, Run. Please clear your App class as per the single responsibility rule.

**Review by tugsem — CHANGES_REQUESTED** [review]
## STATUS: REQUIRED CHANGES ♻️
Hello 👋 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text

 ## Highlights 🚀
✔️Good job on fixing the issues, app working well

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ismailco — APPROVED** [review]
### Hi @Alejuda, @MMGGYY66, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**